lib/backend/grpcclient/utils.go

```diff
@@ -107,8 +107,8 @@ func toGrpcOpType(in types.OpType) grpcbackend.OpType {
 func ConvertGrpcItemsBackendItems(in []*grpcbackend.Item) []backend.Item {
 	result := make([]backend.Item, 0, len(in))
 
-	for _, v := range in {
-		result = append(result, *ConvertGrpcItemBackendItem(v))
+	for i, v := range in {
+		result[i] = *ConvertGrpcItemBackendItem(v)
```

Comment:
This will panic the size of the `result` was set to `0`-  `result := make([]backend.Item, 0, len(in))`

Reply:
Panic for what reason? `make` with a 0 len is legal, and the direct assignment won't happen when `len(in)==0`. 

The `range` is over the length of the input, which is the same as the length of `result`

See https://go.dev/play/p/ZXb4jhByxyS for a contrived example

Reply:
In the provided example the `exampleFunc` is called only with nil and len(array) == 0 args:
```go
exampleFunc(nil)                       //nil input
exampleFunc((make([]*grpcItem, 0, 0))) // functionally the same
```
Please check the modify  example https://go.dev/play/p/AmpQXse2U1p where the input array contains valid elements:
```go
items := []*grpcItem{&grpcItem{input: "foo"}, &grpcItem{input: "bar"}}
exampleFunc(items)
```

Reply:
@smallinsky is correct here. The index needs to be `0 <= x < len(arr)` where on a slice with 0 elements len(arr) is 0. The capacity of the slice might be greater, depending on the input length. So append on the slice will increase the slice length without reallocating, but indexing outside of the slice range will generate an index out of bounds.

https://go.dev/ref/spec#Index_expressions

For the proposed indexing to work, `result := make([]backend.Item, 0, len(in))` would need to change to `result := make([]backend.Item, len(in), len(in))` which would allocate a slice with the required length.

Generally, I use the pattern of initializing the slice/array to the correct size and appending as I'm under the impression that is the approach with the least downsides.